### PR TITLE
feat: allow users to rename drives

### PR DESCRIPTION
**Pull request by jdaev**

### CODE REVIEW

File: lib/blocs/drive_rename/drive_rename_cubit.dart [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        final driveEntity = drive.asEntity();
+
+        final driveTx = await _arweave.prepareEntityTx(
+            driveEntity, profile.wallet, driveKey);
+
+        await _arweave.postTx(driveTx);
+        await _driveDao.writeToDrive(drive);
+
+        driveEntity.txId = driveTx.id;
+
```
Comment (CDDelta) [inline lib/blocs/drive_rename/drive_rename_cubit.dart:85]: You need to set the `ownerAddress` on `driveEntity` here using `profile.address` or something similar to that.

File: lib/blocs/drive_rename/drive_rename_cubit.dart [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    // Check that the current drive does not already have a drive with the target file name.
+    final drivesWithName = (await _driveDao.allDrives().get())
+        .where((element) => element.name == newDriveName);
+    final nameAlreadyExists = drivesWithName.isNotEmpty;
+
+    if (nameAlreadyExists) {
+      control.markAsTouched();
+      return {AppValidationMessage.nameAlreadyPresent: true};
```
Comment (CDDelta) [inline lib/blocs/drive_rename/drive_rename_cubit.dart:113]: We need to add a new `AppValidationMessage` like `driveNameAlreadyPresent`. Also maybe rename `nameAlreadyPresent` to something else like `fsEntryNameAlreadyPresent`.

**Comment by CDDelta** [comment]
LGTM! Works perfectly. Just update the validation message and we should be good to merge.

**Review by agsuy — APPROVED** [review]
Approved. We could rework the UI later on

**Pull request merged**